libraries/redcore/database/driver.php

```diff
@@ -248,5 +247,203 @@ public static function deleteInstances()
 		{
 			self::$instances = null;
 		}
+
+		/**
+		 * Splits a string of multiple queries into an array of individual queries.
+		 *
+		 * @param   string  $sql  Input SQL string with which to split into individual queries.
+		 *
+		 * @return  array  The queries from the input string separated into an array.
+		 *
+		 * @since   11.1
+		 */
+		public static function splitSql($sql)
+		{
+			$start = 0;
+			$open = false;
+			$char = '';
+			$end = strlen($sql);
+			$queries = array();
+			$delimiter = ';';
+			$line = '';
+			$latest = '';
+
+			for ($i = 0; $i < $end; $i++)
+			{
+				$current = substr($sql, $i, 1);
+				$latest .= $current;
+
+				if (strlen($latest) > strlen($delimiter))
+				{
+					$latest = substr($latest, 1);
+				}
+
+				if (($current == '"' || $current == '\''))
+				{
+					$n = 2;
+
+					while (substr($sql, $i - $n + 1, 1) == '\\' && $n < $i)
+					{
+						$n++;
+					}
+
+					if ($n % 2 == 0)
+					{
+						if ($open)
+						{
+							if ($current == $char)
+							{
+								$open = false;
+								$char = '';
+							}
+						}
+						else
+						{
+							$open = true;
+							$char = $current;
+						}
+					}
+				}
+
+				$fileend = ($i == $end - 1);
+
+				if (!$open || $fileend)
+				{
+					$delimiterChange = false;
+
+					if ($current == chr(10))
+					{
+						$statement = substr($sql, $start, ($i - $start + 1));
+
+						if (preg_match('/delimiter(\\s*)(\'?)(.*?)(\'?)$/i', $statement, $matches))
+						{
+							$delimiter = $matches[3];
+							$start = $i + 1;
+							$latest = '';
+							$delimiterChange = true;
+						}
+					}
+
+					if ((!$delimiterChange && $latest == $delimiter) || $fileend)
+					{
+						$queries[] = substr($sql, $start, ($i - $start + 1) - strlen($delimiter));
+						$start = $i + 1;
+						$latest = '';
+					}
+				}
+			}
+
+			return $queries;
+		}
+
+		/**
+		 * This function replaces a string identifier <var>$prefix</var> with the string held is the
+		 * <var>tablePrefix</var> class variable.
+		 *
+		 * @param   string  $sql           The SQL statement to prepare.
+		 * @param   string  $tablePrefix   This install table prefix
```

Comment:
yep, it won't execute it because it's an abstract class (it ends up in an error)
my other option is moving the function to a helper, but I thought this class was a better place so it's in context